### ISSUE TITLE
Add support for 'filter' option

### ISSUE THREAD

**Issue by christophersmith262** [sergei-startsev/ExecutableDllPlugin#2]
## Description

First off, thanks for the geat plugin! It saved me from having to dig into webpack's innards.

I noticed you switched from filtering by `m.id` to `m.identifier()` in cb703a749fd2d6744d4cba783b5a8b0f9018f07d.

I don't pretend to understand the exact difference in the change, but I did want to call out that `m.identifier()` can get pretty difficult to work with in some cases.

For example, here's a dump of `m.identifier()`, taken at random:

```
/Users/smitc2/Sites/example/node_modules/babel-loader/lib/index.js??ref--6!/Users/smitc2/Sites/example/node_modules/@bolt/components-icons/src/icons/twitter.js
```

I'm not sure if this is a result of how we're using babel, but I wanted to call out that comparing by `m.resource` seems to be easier to work with:

```
/Users/smitc2/Sites/example/node_modules/@bolt/components-icons/src/icons/twitter.js
```

## Proposed Solution

As a non-breaking change, I'm suggesting that another option `filter` be added, in addition to the existing `execute` option. This would allow implementations to do more sophisticated filtering on module, eg.:

```js
new ExecutableDllPlugin({
  filter: m => {
    return [path.resolve(__dirname, './src/A.js')].includes(m.resource)
  }
})
```

**Comment by christophersmith262** [comment]
I've raised https://github.com/sergei-startsev/ExecutableDllPlugin/pull/3 to demonstrate my proposed solution.

Also, if you're interested in getting some automated tests set up for this plugin, I did some leg work to write some tests to run against my PR mentioned above:

https://github.com/christophersmith262/ExecutableDllPlugin/tree/feature/add-test-coverage

**Comment by sergei-startsev** [comment]
`m.id` is useless in production mode, see the typical modules ids in prod mode:

`["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10" ... ]`

`m.identifier()` is used in most original webpack plugins, e.g.:

- [PregressPlugin](https://github.com/webpack/webpack/blob/0293c3a5ee6e41d04f4c5e1fb0607fdb7a0b38a8/lib/ProgressPlugin.js#L174)
- [SourceMapDevToolsPlugin](https://github.com/webpack/webpack/blob/3b6d1495c3858d59cfdb46b2343c07f2e4c1316d/lib/SourceMapDevToolPlugin.js#L174)
- [CaseSensitiveModulesWarning](https://github.com/webpack/webpack/blob/8a7597aa6eb2eef66a8f9db3a0c49bcb96022a94/lib/CaseSensitiveModulesWarning.js#L17)
- ...

And it's used by this plugin as well. Anyway having the custom filter allows you to implement more sophisticated algorithm. That's what was planned to be added initially.

BTW
`Users/smitc2/Sites/example/node_modules/babel-loader/lib/index.js??ref--6!/Users/smitc2/Sites/example/node_modules/@bolt/components-icons/src/icons/twitter.js`

I don't think you actually need to transpile your `node_modules`, see [babel-loader is slow](https://webpack.js.org/loaders/babel-loader/#babel-loader-is-slow).

**Comment by sergei-startsev** [comment]
The issue was fixed in 51d78fca3cafe3d47ff701028a63c769a5450111
